electrumsv/wallet_database/functions.py

```diff
@@ -640,9 +640,10 @@ def read_history_for_outputs(db: sqlite3.Connection, account_id: int, *,
     INNER JOIN TransactionOutputs TXO ON TXO.tx_hash=ATX.tx_hash
     LEFT JOIN _AccountSpends _AS ON _AS.tx_hash=ATX.tx_hash
     LEFT JOIN KeyInstances KI ON KI.keyinstance_id=TXO.keyinstance_id
+    --TXO.keyinstance_id IS NULL indicates that it's paying an external wallet
     WHERE {transaction_clause_and} ((_AS.spent_value>0 AND (TXO.keyinstance_id IS NULL OR
         KI.derivation_type=={DerivationType.BIP32_SUBPATH} AND KI.derivation_data2<x'00000001'
-        AND KI.derivation_data2>=x'00000002'))
+        OR KI.derivation_data2>=x'00000002'))
```

Comment:
I have no idea why you have made this change or what it is supposed to accomplish.

Reply:
`KI.derivation_data2<x'00000001' AND KI.derivation_data2>=x'00000002'` doesn't make any sense... The intent is to exclude keys used for change right? But if you think about it, it's not possible to ever satisfy this condition.

It should be OR not AND